File: src/main/scala/com/wanda/blockchain/common/chaincode/ChainCodeInterface.scala
package com.wanda.blockchain.common.chaincode

/**
  * Created by Zhou peiwen on 2017/7/4.
  */
trait ChainCodeInterface {

  def setHandler(cCHandlerInterface: CCHandlerInterface):Unit

  def getHandler:CCHandlerInterface

  def init:Unit

  def invoke(method:String,param:java.util.ArrayList[String]):CCResponse

}

class CCResponse(val res:java.util.HashMap[String,Any])

File: src/main/java/com/wanda/blockchain/common/chaincode/ChainCodeInterface.java
package com.wanda.blockchain.common.chaincode;

import java.util.ArrayList;

/**
 * Created by Zhou peiwen on 2017/7/10.
 */
public interface ChainCodeInterface {

    public void setHandler(CCHandlerInterface handler);

    public CCHandlerInterface getHandler();

    public void init();

    public CCResponse invoke(String method,ArrayList<String> param);
}
